### DISCUSSION TITLE
Entity ID by DeviceID

### DISCUSSION THREAD

**Discussion by vzabavnov** [net-daemon/netdaemon#1085] [Q&A]
Hello All,
I have many Z-Wave devices and all of them report event's data with Device ID. I need somehow map Device ID in event data to Entity ID. Currently I do it manually. but will be nice to have some kind of call to get device id by entity id:
```C#
IHaContect haContext = ...;

string deviceID = haContext.GetDeviceID("light.room_bed_upper_light")
```
there is HA code the returns what I need:
```
{{device_id('sensor.mysensor')}}
```
Anyone know how to call it from NetDaemon's context?

**Comment by helto4real** [comment]
If there are a limited number and it does not need to be dynamic I figure you can check out the templated sensor.

Another way would be to create a script that takes data. I am sure you can google how that is done. You will have to use `CallService` directly since we currently does not support providing data to code generated scripts.

**Reply by helto4real** [reply]
For faster answers please join our Discord server.
https://discord.gg/K3xwfcX
